src/main/java/com/sequenceiq/cloudbreak/logger/MDCBuilder.java

```diff
@@ -32,29 +16,37 @@ public static void buildMdcContext() {
 
     public static void buildMdcContext(Object object) {
         if (object == null) {
-            CloudBreakLoggerFactory.buildMdcContext();
-            return;
-        }
-        if (object instanceof Credential) {
-            CredentialLoggerFactory.buildMdcContext((Credential) object);
-        } else if (object instanceof Stack) {
-            StackLoggerFactory.buildMdcContext((Stack) object);
-        } else if (object instanceof Cluster) {
-            ClusterLoggerFactory.buildMdcContext((Cluster) object);
-        } else if (object instanceof Template) {
-            TemplateLoggerFactory.buildMdcContext((Template) object);
-        } else if (object instanceof Blueprint) {
-            BlueprintLoggerFactory.buildMdcContext((Blueprint) object);
-        } else if (object instanceof Recipe) {
-            RecipeLoggerFactory.buildMdcContext((Recipe) object);
-        } else if (object instanceof CloudbreakEvent) {
-            EventLoggerFactory.buildMdcContext((CloudbreakEvent) object);
-        } else if (object instanceof CloudbreakUsage) {
-            UsageLoggerFactory.buildMdcContext((CloudbreakUsage) object);
-        } else if (object instanceof CloudbreakEventData) {
-            EventDataLoggerFactory.buildMdcContext((CloudbreakEventData) object);
+            MDC.put(LoggerContextKey.OWNER_ID.toString(), "cloudbreak");
+            MDC.put(LoggerContextKey.RESOURCE_TYPE.toString(), "cloudbreakLog");
+            MDC.put(LoggerContextKey.RESOURCE_ID.toString(), "undefined");
+            MDC.put(LoggerContextKey.RESOURCE_NAME.toString(), "cb");
         } else {
-            throw new UnsupportedOperationException(String.format("%s class not supported for logging.", object.getClass()));
+            try {
+                Field privateStringField = object.getClass().getDeclaredField("owner");
+                privateStringField.setAccessible(true);
+                MDC.put(LoggerContextKey.OWNER_ID.toString(), (String) privateStringField.get(object));
+            } catch (Exception e) {
+                MDC.put(LoggerContextKey.OWNER_ID.toString(), "undefined");
+            }
+            try {
+                MDC.put(LoggerContextKey.RESOURCE_TYPE.toString(), object.getClass().getSimpleName().toUpperCase());
+            } catch (Exception e) {
+                MDC.put(LoggerContextKey.RESOURCE_TYPE.toString(), "undefined");
+            }
+            try {
+                Field privateStringField = object.getClass().getDeclaredField("name");
+                privateStringField.setAccessible(true);
+                MDC.put(LoggerContextKey.RESOURCE_NAME.toString(), (String) privateStringField.get(object));
+            } catch (Exception e) {
+                MDC.put(LoggerContextKey.RESOURCE_NAME.toString(), "undefined");
+            }
+            try {
```

Comment:
We can refactor these try catches to 1 method which takes the declaredfield and LoggerContextKey which return the desired value or undefined. At least 3 segment can be reduced.